### ISSUE TITLE
needless_pass_by_value false positive with String

### ISSUE THREAD

**Issue by steveklabnik** [rust-lang-nursery/rust-clippy#2114]
this code:

```rust
fn foo(s: String) {
    s.capacity()
}
```

triggers `needless_pass_by_value`, but I cannot turn this into `&str`, since there's no capacity method on it.

**Comment by llogiq** [comment]
Strange, this should have been caught by #2079 – are you on the newest clippy? If so, I shall investigate.

**Comment by steveklabnik** [comment]
I did this on the playpen; unsure what version it's using.

**Comment by llogiq** [comment]
It appears to be a slightly outdated Version then. At least I cannot reprodurce with current clippy.

**Comment by sinkuu** [comment]
#2079 is for ptr_arg. I'll duplicate ptr_arg's logic for needless_pass_by_value.
